RIOT-OS/RIOT
# make: remove exports for flash debug reset

### Contribution description

This removes `export` for all variables related to `flash/reset/debug`.
They do not need to be exported as they are all evaluated in `Makefile.include` or a file included by it.

This does not remove export `PORT` for the moment as it is easier to review it on its own as it will have a different review scheme

### Testing procedure

There is not real testing procedure except, it would all still be working, but not really realistic for testing.


Better do an in depth review to find issues.
If any issue happened because of this, it could be easier to fix the missing case.


Running `make flash/make reset/make debug` still works on one of your boards.
At least it should not break `CI`.

### Review prodecure 

The only usages of the variables are in `Makefile.include` or `.inc.mk` files:

```
git grep -e '$(FFLAGS)' -e '$(FLASHER)' -e '$(RESET)' -e '$(RESET_FLAGS)' -e '$(DEBUGGER)' -e '$(DEBUGGER_FLAGS)' -e '$(DEBUGSERVER)' -e '$(DEBUGSERVER_FLAGS)' -e '$(MSPDEBUGFLAGS)'
Makefile.include:  $(call check_cmd,$(FLASHER),Flash program)
Makefile.include:  $(FLASHER) $(FFLAGS)
Makefile.include:       $(call check_cmd,$(DEBUGGER),Debug program)
Makefile.include:       $(DEBUGGER) $(DEBUGGER_FLAGS)
Makefile.include:       $(call check_cmd,$(DEBUGSERVER),Debug server program)
Makefile.include:       $(DEBUGSERVER) $(DEBUGSERVER_FLAGS)
Makefile.include:       $(call check_cmd,$(RESET),Reset program)
Makefile.include:       $(RESET) $(RESET_FLAGS)
boards/common/msb-430/Makefile.include:FFLAGS = $(MSPDEBUGFLAGS) "prog $(HEXFILE)"
boards/common/msb-430/Makefile.include:DEBUGSERVER = $(FLASHER)
boards/common/msb-430/Makefile.include:DEBUGSERVER_FLAGS = $(MSPDEBUGFLAGS) gdb
boards/common/msba2/Makefile.include:FLASHDEPS += $(if $(findstring $(LPC2K_PGM),$(FLASHER)),$(LPC2K_PGM))
boards/native/Makefile.include:ifeq ($(shell basename $(DEBUGGER)),lldb)
boards/native/Makefile.include: $(eval DEBUGGER_FLAGS := -ex "target remote | vgdb --pid=$(VALGRIND_PID)" $(DEBUGGER_FLAGS))
boards/native/Makefile.include: $(DEBUGGER) $(DEBUGGER_FLAGS)
boards/teensy31/Makefile.include:ifeq ($(TEENSY_LOADER),$(FLASHER))
makefiles/info.inc.mk:  @echo 'FLASHER: $(FLASHER)'
makefiles/info.inc.mk:  @echo 'FFLAGS:  $(FFLAGS)'
makefiles/info.inc.mk:  @echo 'DEBUGGER:       $(DEBUGGER)'
makefiles/info.inc.mk:  @echo 'DEBUGGER_FLAGS: $(DEBUGGER_FLAGS)'
makefiles/info.inc.mk:  @echo 'DEBUGSERVER:       $(DEBUGSERVER)'
makefiles/info.inc.mk:  @echo 'DEBUGSERVER_FLAGS: $(DEBUGSERVER_FLAGS)'
makefiles/info.inc.mk:  @echo 'RESET:       $(RESET)'
makefiles/info.inc.mk:  @echo 'RESET_FLAGS: $(RESET_FLAGS)'
makefiles/murdock.inc.mk:FLASHFILE ?= $(filter $(HEXFILE) $(ELFFILE:.elf=.bin) $(ELFFILE),$(FFLAGS))
makefiles/tools/avrdude.inc.mk:DEBUGGER = $(DIST_PATH)/debug.sh $(DEBUGSERVER_FLAGS) $(DIST_PATH) $(DEBUGSERVER_PORT)
makefiles/tools/bossa.inc.mk:ifeq ($(RIOTTOOLS)/bossa/bossac,$(FLASHER))
makefiles/tools/edbg.inc.mk:ifeq ($(RIOT_EDBG),$(FLASHER))
```

No match when using `${}`
```
git grep -e '${FFLAGS}' -e '${FLASHER}' -e '${RESET}' -e '${RESET_FLAGS}' -e '${DEBUGGER}' -e '${DEBUGGER_FLAGS}' -e '${DEBUGSERVER}' -e '${DEBUGSERVER_FLAGS}' -e '${MSPDEBUGFLAGS}'
```

#### Only required changes are changed

The only changes are `[-export-]` and comments `{+#+}`

```git diff --color=auto --word-diff 1dc479de0 '**' ':!dist/tools/buildsystem_sanity_check/check.sh'```


Only the right amount of lines have been changed

```
git diff  --stat  1dc479de0 '**' ':!dist/tools/buildsystem_sanity_check/check.sh'
...
 29 files changed, 181 insertions(+), 181 deletions(-)
```
*Edited to 29 files changed*.

This includes:

* 170 `export` changed.
* 8 modifications in `makefiles/vars.inc.mk`
* 2 lines for `MSPDEBUGFLAGS` (that I did not added to build_system_check as really specific)
* 1 newline that is reported in `cpu/esp32/Makefile.include` for `PREFFFLAGS`


And from the `buildsystem_sanity_check` output, on the original commit 1dc479de0

```
git checkout d486598fdf3c72b7b110ff97a9ea2dbb28f50e1f -- dist/tools/buildsystem_sanity_check/check.sh
./dist/tools/buildsystem_sanity_check/check.sh  | grep -v 'Invalid build system patterns found by' | wc -l
178
```
> *Edited from 170 to 178 now that `makefiles/vars.inc.mk` is also parsed*

Matches the number of `export` changed


### Issues/PRs references

Part of https://github.com/RIOT-OS/RIOT/issues/10850
In depth cleanup required for https://github.com/RIOT-OS/RIOT/pull/10440
